### PR TITLE
GET/POST API changes for cves additional property

### PR DESCRIPTION
**Pull request by rajusem**
API changes for saving and retrieving cves additional property

### CODE REVIEW

File: src/rest_model.py [hunk 1]
```diff
@@ -21,7 +21,8 @@
     'closed_at': fields.DateTime(dt_format='iso8601'),
     'creator_name': fields.String,
     'creator_url': fields.Url(description='Creator Url'),
-    'probable_cve': fields.Boolean()
+    'probable_cve': fields.Boolean(),
+    'cves': fields.List(fields.String, description='Un-verified CVEs got from title/body')
```
Comment (arajkumar) [inline src/rest_model.py]:
```suggestion
    'cves': fields.List(fields.String, description='Un-verified CVE list parsed from title and body')
```

File: src/graph_model.py [hunk 2]
```diff
@@ -129,3 +129,4 @@ class SecurityEvent(BaseModel):
     updated_date: int  # yyyyMMdd format
     updated_yearmonth: int  # yyyyMM format
     updated_year: int  # yyyy format
+    cves: list
```
Comment (arajkumar) [inline src/graph_model.py]:
```suggestion
    cves: List[str]
```

File: src/rest_model.py [hunk 3]
```diff
@@ -75,8 +76,8 @@ def _check_overall_feedback(x):
                                     description='Title for the issue/pr'),
     'body': UnsanitizeStringFormat(attribute=lambda x: None if 'body' not in x else x['body'][0],
                                    description='Description for the issue/pr'),
+    'cves': fields.List(fields.String, description='Un-verified CVEs got from title/body'),
```
Comment (arajkumar) [inline src/rest_model.py:78]:
```suggestion
    'cves': fields.List(fields.String, description='Un-verified CVE list parsed from title and body'),
```

File: src/ingestion_data.py [hunk 4]
```diff
@@ -4,6 +4,9 @@
 
 from src.graph_model import (Version, EventType, SecurityEvent, FeedBackType, StatusType, EcosystemType)
 from src.parse_datetime import from_date_str, get_date, get_year, get_yearmonth
+from src.config import MAX_STRING_LENGTH
+
+from src.utils import get_cves_from_text
```
Comment (arajkumar) [inline src/ingestion_data.py:9]: I suggest to keep `get_cves_from_text` local to this package as I don't anticipate any use outside of this.

File: src/utils.py [hunk 5]
```diff
@@ -42,3 +44,9 @@ def fetch_nodes(payload: object) -> object:
             raise e
     else:
         return {'warning': 'Invalid payload. Check your payload once again'}
+
+
+def get_cves_from_text(data: str) -> set:
```
Comment (arajkumar) [inline src/utils.py:49]:
```suggestion
def get_cves_from_text(data: str) -> Set[str]:
```

File: src/utils.py [hunk 6]
```diff
@@ -2,10 +2,12 @@
 
 import daiquiri
 import requests
+import re
 
 from src.config import DAIQUIRI_LOG_LEVEL
 from src.gremlin import execute_query
 
+CVE_REGULAR_EXPRESSION = r"CVE-\d{4}-\d{4,}"
```
Comment (arajkumar) [inline src/utils.py:10]: Are you sure that second part will be always 4 digits? `\d{4}`

Comment (rajusem) [reply src/utils.py:10]: Its `\d{4,}` so 4 or more digit we are expecting.

File: src/graph_model.py (hunk 2 above)
Comment (rajusem) [reply src/graph_model.py]: outdated

File: src/utils.py (hunk 5 above)
Comment (rajusem) [reply src/utils.py]: As discussed not changing it to Set

File: src/ingestion_data.py (hunk 4 above)
Comment (rajusem) [reply src/ingestion_data.py]: moved to same file

File: src/rest_model.py (hunk 3 above)
Comment (rajusem) [reply src/rest_model.py]: done

File: src/ingestion_data.py [hunk 7]
```diff
@@ -16,6 +25,16 @@ def __init__(self, json_data):
         self._payload = json_data
         self._sec = None
         self._update_sec = None
+        self._cves = self._get_cves()
+        if self._payload['body']:
+            self._payload['body'] = self._payload['body'][0:MAX_STRING_LENGTH]
+
+    def _get_cves(self):
+        """Get the CVEs mentioned from title or body."""
+        txt = self._payload['title']
```
Comment (arajkumar) [inline src/ingestion_data.py:34]:
```suggestion
        title = self._payload['title']
```

File: src/ingestion_data.py [hunk 8]
```diff
@@ -4,6 +4,15 @@
 
 from src.graph_model import (Version, EventType, SecurityEvent, FeedBackType, StatusType, EcosystemType)
 from src.parse_datetime import from_date_str, get_date, get_year, get_yearmonth
+from src.config import MAX_STRING_LENGTH
+
+CVE_REGULAR_EXPRESSION = r"CVE-\d{4}-\d{4,}"
```
Comment (arajkumar) [inline src/ingestion_data.py:9]: This should be private constant.
```suggestion
_CVE_REGULAR_EXPRESSION = r"CVE-\d{4}-\d{4,}"
```

File: src/ingestion_data.py [hunk 9]
```diff
@@ -4,6 +4,15 @@
...
+
+
+def get_cves_from_text(data: str) -> set:
+    """Get the CVEs from the given text."""
+    cves = re.findall(CVE_REGULAR_EXPRESSION, data.upper())
+    return set(cves)
```
Comment (arajkumar) [inline src/ingestion_data.py:15]: Return it as a list and a set can be formulated from the list from the caller side.

File: src/ingestion_data.py [hunk 10]
```diff
@@ -4,6 +4,15 @@
 
 from src.graph_model import (Version, EventType, SecurityEvent, FeedBackType, StatusType, EcosystemType)
 from src.parse_datetime import from_date_str, get_date, get_year, get_yearmonth
+from src.config import MAX_STRING_LENGTH
+
+CVE_REGULAR_EXPRESSION = r"CVE-\d{4}-\d{4,}"
+
+
+def get_cves_from_text(data: str) -> set:
```
Comment (arajkumar) [inline src/ingestion_data.py:12]: This should be a private function.
```suggestion
def _get_cves_from_text(data: str) -> set:
```

File: src/ingestion_data.py [hunk 11]
```diff
@@ -16,6 +25,16 @@ def __init__(self, json_data):
...
+        self._cves = self._get_cves()
+        if self._payload['body']:
+            self._payload['body'] = self._payload['body'][0:MAX_STRING_LENGTH]
+
+    def _get_cves(self):
+        """Get the CVEs mentioned from title or body."""
+        txt = self._payload['title']
+        if self._payload['body']:
+            txt = txt + self._payload['body']
+        return get_cves_from_text(txt)
```
Comment (arajkumar) [inline src/ingestion_data.py:37]: Usually the large text concat op would be expensive in terms of memory & CPU. In this you can do isolated text extraction.
```suggestion
        cves = get_cves_from_text (self._payload['title'])
        if self._payload['body']:
            cves.append(get_cves_from_text(self._payload['body']))
        return set(cves)
```

File: src/ingestion_data.py (hunk 7 above)
Comment (rajusem) [reply src/ingestion_data.py:34]: As same variable we are appending body as well, making `title` don't seems to right.

File: src/ingestion_data.py (hunk 11 above)
Comment (rajusem) [reply src/ingestion_data.py]: done

File: src/ingestion_data.py (hunk 10 above)
Comment (rajusem) [reply src/ingestion_data.py]: done

File: src/ingestion_data.py (hunk 9 above)
Comment (rajusem) [reply src/ingestion_data.py]: done

File: src/ingestion_data.py (hunk 8 above)
Comment (rajusem) [reply src/ingestion_data.py]: done

File: src/utils.py (hunk 6 above)
Comment (arajkumar) [reply src/utils.py]: In the format CVE-X-Y, I had mentioned about the X part. So the X always denotes year?

**Pull request merged**